src/test/scala/org/constellation/p2p/PeerAPITest.scala

```diff
@@ -85,24 +97,109 @@ class PeerAPITest
+        status shouldEqual StatusCodes.OK
+        responseAs[Array[Byte]] shouldBe Array.empty[Byte]
+      }
+    }
+  }
+
+  "POST snapshot/info/height" - {
+    "response should return response with all own snapshots at given height" in {
+      val storedSnapshot = SnapshotInfo(StoredSnapshot(Snapshot("hash", Seq.empty, Map.empty), Seq()))
+      storedSnapshot.writeLocal(snapshotWritePath.pathAsString)
+      val heightToReturn = 0L
+      val pd = mock[PeerData]
+      pd.peerMetadata shouldReturn mock[PeerMetadata]
+      pd.peerMetadata.id shouldReturn Id("foo")
+      dao.cluster.getPeerData("localhost") shouldReturnF Some(pd)
+      DateTimeUtils.setCurrentMillisFixed(1234567)
+      dao.snapshotInfoPath shouldReturn snapshotWritePath
+      dao.snapshotService shouldReturn mock[SnapshotService[IO]]
+      dao.rollbackLoader shouldReturn mock[RollbackLoader]
+      dao.snapshotInfoPath shouldReturn snapshotWritePath
+      dao.snapshotService.recentSnapshotInfo shouldReturn mock[StorageService[IO, SnapshotInfoSer]]
+      dao.snapshotService.recentSnapshotInfo.put(*, *) shouldReturnF storedSnapshot.toSnapshotInfoSer()
+
+      Post(s"/snapshot/info/${heightToReturn}", ("address", Array.empty[Array[Byte]])) ~> peerAPI.postEndpoints(socketAddress) ~> check {
+        dao.snapshotService.recentSnapshotInfo.put(*, *).was(called)
+        status shouldEqual StatusCodes.OK
+        responseAs[Array[Byte]] shouldBe Array.empty[Array[Byte]]
+      }
+    }
+
+    "response should not return if below max height" in {
+      val storedSnapshot = SnapshotInfo(StoredSnapshot(Snapshot("hash", Seq.empty, Map.empty), Seq()))
+      storedSnapshot.writeLocal(snapshotWritePath.pathAsString)
+      val heightToReturn = 2L
+      val pd = mock[PeerData]
+      pd.peerMetadata shouldReturn mock[PeerMetadata]
+      pd.peerMetadata.id shouldReturn Id("foo")
+      dao.cluster.getPeerData("localhost") shouldReturnF Some(pd)
+      DateTimeUtils.setCurrentMillisFixed(1234567)
+      dao.snapshotInfoPath shouldReturn snapshotWritePath
+      dao.snapshotService shouldReturn mock[SnapshotService[IO]]
+      dao.rollbackLoader shouldReturn mock[RollbackLoader]
+      dao.snapshotService.storedSnapshot.get shouldReturnF mock[StoredSnapshot]
+      dao.snapshotService.storedSnapshot.get.unsafeRunSync().height shouldReturn 0L
+      dao.snapshotInfoPath shouldReturn snapshotWritePath
+      dao.snapshotService.recentSnapshotInfo shouldReturn mock[StorageService[IO, SnapshotInfoSer]]
+      dao.snapshotService.recentSnapshotInfo.put(*, *) shouldReturnF storedSnapshot.toSnapshotInfoSer()
+
+      Post(s"/snapshot/info/${heightToReturn}", ("address", Array.empty[Array[Byte]])) ~> peerAPI.postEndpoints(socketAddress) ~> check {
+        dao.snapshotService.recentSnapshotInfo.put(*, *).wasNever(called)
+        status shouldEqual StatusCodes.NotFound
       }
     }
   }
+  override def afterAll {
+    File(parentDir).createDirectoryIfNotExists().delete()
```

Comment:
Hacky and ducttaped imo ;/
No need to create dir just to delete it immediately.

Either swallow exception:
```scala
File(parentDir).delete(swallowIOExceptions = true)
```

Or just check if exists:

```scala
val file = File(parentDir)
if (file.exists) { file.delete())
```

Reply:
Yep, we gotta remove all those too cause of the file descriptors leak